docs/source/concepts/settlement.rst

```diff
@@ -7,38 +7,147 @@ Settlement
 
-Step
-****
+We then delve into the details of each of the settlement components.
 
-The FX example transaction above contains two steps:
+Workflow
+********
 
-#. transfer EUR from Alice to Bob
-#. transfer USD from Bob to Alice
+Our initial state looks as follows:
 
-They are represented using one ``Step`` each.
-The step defines who is the sender, who is the receiver and what should be transferred (instrument and amount).
+* Alice owns a holding on an `EUR` instrument, for an amount of `1000`
+* Bob owns a holding on a `USD` instrument, for an amount of `1000`
+
+These holding are generally held at different custodians.
+
+.. image:: ../images/settlement_initial_state.png
+   :alt: Alice owns a EUR holding, Bob owns a USD holding.
+
+Instruct
+========
+
+Alice and Bob want to exchange their holdings and agree to enter into the transaction by being signatories on a transaction contract.
+Settlement can then be instructed: this generates
+
+#. an ``Instruction`` to transfer 1000 EUR from Alice to Bob
+#. an ``Instruction`` to transfer 1000 USD from Bob to Alice
+#. a ``Batch`` contract which is used to execute the above Instructions.
+
+.. image:: ../images/settlement_instructed.png
+   :alt: Settlement is instructed.
+
+Each instruction defines, for a step of the settlement chain, who is the sender, who is the receiver and what should be transferred (instrument and amount).
+
+Allocate and Approve
+====================
+
+In order to perform the transfer, we now need to specify for each ``Instruction``
+
+#. which holding should be used
+#. to which account it should be transferred.
+
+Alice and Bob ``allocate`` the instruction where they are, respectively, the sender by pledging their holding.
+
+.. image:: ../images/settlement_allocated.png
+   :alt: Settlement is allocated.
+
+Each receiver can then specify to which account the holding should be sent to by ``approving`` the corresponding instruction.
+
+.. image:: ../images/settlement_allocated_approved.png
+   :alt: Settlement is allocated and approved.
+
+Execute
+=======
+
+Once instructions are allocated and approved, a Settler party uses the ``Batch`` contract to ``execute`` them and finalize settlement in one atomic transaction.
+
+.. image:: ../images/settlement_executed.png
+   :alt: Settlement is allocated and approved.
+
+The instructions and batch are archived following a successful execution.
+
+Remarks
+=======
+
+There are some assumptions that need to be verified in order for the transaction above to settle successfully in practice. On the first instruction
+
+- Bob needs to have an account at the custodian where Alice's holding is held
+- Alice's holding needs to be :ref:`Transferable <type-daml-finance-interface-holding-transferable-transferable-24986>`
+- The transfer must be fully authorized (TODO : add a link to a page where we describe the role of account instructors and approvers, probably somewhere in the asset model)
+
```

Comment:
Do you want to fix the TODO in this or in a separate PR?

Reply:
Yes, I raised https://github.com/digital-asset/daml-finance/issues/330 for it yesterday. Will remove the TODO from the code